src/targets.rs

```diff
@@ -96,6 +96,63 @@ impl Default for InitializationConfig {
     }
 }
 
+#[derive(Eq)]
+pub struct TargetTriple {
+    pub(crate) target_triple: LLVMStringOrRaw,
+}
+
+impl TargetTriple {
+    pub(crate) fn new_owned(target_triple: *const ::libc::c_char) -> TargetTriple {
+        debug_assert!(!target_triple.is_null());
+
+        TargetTriple {
+            target_triple: LLVMStringOrRaw::Owned(LLVMString::new(target_triple)),
+        }
+    }
+
+    pub(crate) fn new_borrowed(target_triple: *const ::libc::c_char) -> TargetTriple {
+        debug_assert!(!target_triple.is_null());
+
+        TargetTriple {
+            target_triple: LLVMStringOrRaw::Borrowed(target_triple),
+        }
+    }
+
+    pub fn create(target_triple: &str) -> TargetTriple {
+        let c_string = CString::new(target_triple).expect("Conversion to CString failed unexpectedly");
```

Comment:
Nit: This will allocate twice, I think? You should just use `create_from_str` on `target_triple`?

Reply:
My thinking was this:
What if the `&str` does not end with a `'\0'`? In general I would not expect it to. So, I would need to append `'\0'` to the input. To append `'\0'` I would need an owned copy, which would reallocate anyway. Am I missing something here?

Reply:
You're right, but there's gotta be a better way..

Reply:
This is abstraction breaking so probably best to avoid, but if you look at the LLVM source code you can see that creating a message just does a `strdup` and freeing it just does a `free` so you could actually pretended any old owned cstring was allocated by LLVM.